feature: support gigachat llm

Rai220 opened the pull request:
Adds gigachat LLM support (from langchain_gigachat package - https://github.com/ai-forever/langchain-gigachat)

virattt commented on src/llm/models.py:
```diff
@@ -188,3 +190,5 @@ def get_model(model_name: str, model_provider: ModelProvider, api_keys: dict = N
             }
         )
+    elif model_provider == ModelProvider.GIGACHAT:
+        return GigaChat(model=model_name)
```
Can we explicitly pass in the API key here?

And add support for being able to pass in the API key from the full-stack app (can use OpenAI as example above)

virattt replied:
Understood

virattt commented on .env.example:
```diff
@@ -22,5 +22,8 @@ FINANCIAL_DATASETS_API_KEY=your-financial-datasets-api-key
 OPENAI_API_KEY=your-openai-api-key
 
+# For running LLM GigaChat
+GIGACHAT_CREDENTIALS=your-gigachat-credentials
```
Let's keep the naming consistent (eg GIGACHAT_API_KEY) and explicitly grab the api key (see comment below)

virattt reviewed (commented):
Left a few comments, but overall aligned!

virattt commented on app/frontend/src/components/settings/api-keys.tsx:
```diff
@@ -65,6 +65,13 @@ const LLM_API_KEYS: ApiKey[] = [
+    key: 'GIGACHAT_API_KEY',
+    label: 'GigaChat API',
+    description: 'For GigaChat models (GigaChat-2-Max, etc.)',
+    url: 'https://github.com/ai-forever/gigachat',
```
Is this the URL to get an API key / credentials?

Rai220 replied:
Yes, this is start point for developers and main pthon lib to use GigaChat. There are links to quickstart for individuals and legal entities.

Rai220 commented:
@virattt Hey, just a quick nudge on PR #414 — I've addressed all your comments. Could you please take a look and consider merging when you get a chance? Thanks! 🙌
